File: scala/wordcount-df.scala
import org.apache.spark.sql.SparkSession

object wordcount {
    def main(args: Array[String]) {
        val file = "1000000.csv"
        val spark: SparkSession = SparkSession.builder
                                              .appName("Scala WordCount DF")
                                              .master("spark://192.168.31.136:7077")
                                              .getOrCreate()

        val df = spark.read.csv(file)
        // df.show(10,truncate=false)
        
        val wordcount = df.groupBy("_c0").count()
        wordcount.show(20, truncate=false)
        
        spark.stop()
    }
}

File: wordcount-df.py
import sys

from pyspark.sql import SparkSession

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: wordcount <file>", file=sys.stderr)
        sys.exit(-1)

    spark = SparkSession\
        .builder\
        .appName("Python DataFrame WordCount")\
        .getOrCreate()

    dataframe = spark.read.csv(sys.argv[1])
    #dataframe.printSchema()
    #dataframe.show(truncate=False)

    wordcount = dataframe.groupBy("_c0").count()
    #wordcount.printSchema()
    wordcount.show(truncate=False)

    spark.stop()